tests/unit/store/adapter-interop-test.js

```diff
@@ -109,6 +109,65 @@ test('Calling Store#findRecord multiple times coalesces the calls into a adapter
   });
 });
 
+test('Coalesced Store#findRecord requests retain the `include` adapter option in the snapshots passed to adapter#findMany', function(assert) {
+  const includedResourcesForIds = {
+    1: 'someResource',
+    2: 'differentResource',
+    3: 'anotherResource'
+  };
+
+  assert.expect(Object.entries(includedResourcesForIds).length);
+
+  const Adapter = TestAdapter.extend({
+    groupRecordsForFindMany(store, snapshots) {
+      // this ensures that `findRecord` is also called
+      return [snapshots.slice(0, -1), snapshots.slice(-1)];
+    },
+
+    findMany(store, type, ids, snapshots) {
+      snapshots.forEach(snapshot => {
+        assert.equal(
+          snapshot.include,
+          includedResourcesForIds[snapshot.id],
+          `Snapshot #${snapshot.id} retains the 'include' adapter option in #findMany`
+        );
+      });
+
+      return resolve({
+        data: snapshots.map(({ id }) => ({ id, type: type.modelName }))
+      });
+    },
+
+    findRecord(store, type, id, snapshot) {
+      assert.equal(
+        snapshot.include,
+        includedResourcesForIds[snapshot.id],
+        `Snapshot #${snapshot.id} retains the 'include' adapter option in #findRecord`
+      );
```

Comment:
Depending on how the actual groupRecordsForFindMany method is implemented, it's possible that a single record of a coalesced group is retrieved via findRecord. In order to assert that findRecord works correctly as well, I've added a custom groupRecordsForFindMany method and added an assertion in here as well.